README.md

```diff
@@ -1,24 +1,74 @@
-# Cartesi Rollups
+# Smart Contracts for Cartesi Rollups
 
-This repository contains the on-chain and off-chain pieces that are used to deploy, launch and interact with Cartesi Rollups DApps. The code presented here is work in progress, continuously being improved and updated.
+This repository contains the on-chain part of Cartesi Rollups.
+
+If you are interested in taking a look at the off-chain part, please, head over to [`cartesi/rollups-node`](https://github.com/cartesi/rollups-node).
+
+## Dependencies
+
+- [Yarn](https://classic.yarnpkg.com/en/docs/install)
+- [Forge](https://book.getfoundry.sh/getting-started/installation)
+- [Docker](https://docs.docker.com/get-docker/)
+
+## Basic setup
+
+This repository contains [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules).
+In order to properly initialize them, please, run the following command.
+
+```sh
+git submodule update --init --recursive
+```
+
+This repository uses [Yarn](https://classic.yarnpkg.com/en/docs/install) to manage JavaScript dependencies.
+In order to install them, please, run the following commands.
+
+```sh
+cd onchain
+yarn install
+```
+
+## Tests
+
+If you plan to run the [Forge](https://book.getfoundry.sh/getting-started/installation) tests, there still some setup left to do.
+Assuming you are on the `onchain` directory, and that [Docker Engine](https://docs.docker.com/get-docker/) is running on the background, you may run the following commands.
+
+```sh
+cd rollups
+yarn proofs:setup
+```
+
+Now, you may run the tests!
+Assuming you are on the `onchain/rollups` directory, please, run the following command.
+
+```sh
+yarn test
+```
+
+From now on, after any change in the source code, you may need to update the proofs used in the tests.
+Assuming you are on the `onchain/rollups` directory, you would do this by running the following command.
+
+```sh
+yarn proofs:update
+```
 
 ## Table of contents
 
 - [Documentation](#documentation)
 - [Experimenting](#experimenting)
 - [Talk with us](#talk-with-us)
 - [Contributing](#contributing)
-- [Setting up](#setting-up)
-- [Testing](#testing)
 - [License](#license)
```

Comment:
Do you think it looks a bit odd if the table of contents is in the middle of a page? Should we change it back similar to what it was to the beginning of a page?

Reply:
Ok, let me fix that real quick.

Reply:
This has been addressed by 039c16b.